src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -4515,11 +4515,21 @@ class MavlinkStreamDistanceSensor : public MavlinkStream
 
 	unsigned get_size() override
 	{
-		return _distance_sensor_sub.advertised() ? (MAVLINK_MSG_ID_DISTANCE_SENSOR_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES) : 0;
+		unsigned total_size = 0;
```

Comment:
In `uORB::SubscriptionMultiArray` there's a method to get the number of advertised.

``` C++
    return size_per_instance * _distance_sensor_subs.advertised_count();
```

Reply:
I'll fix this.